node/src/components/network/event.rs

```diff
@@ -0,0 +1,176 @@
+use std::{
+    fmt::{self, Debug, Display, Formatter},
+    io,
+    num::NonZeroU32,
+};
+
+use derive_more::From;
+use libp2p::{
+    core::connection::{ConnectedPoint, PendingConnectionError},
+    Multiaddr,
+};
+use serde::Serialize;
+
+use super::Message;
+use crate::{
+    effect::requests::{NetworkInfoRequest, NetworkRequest},
+    types::NodeId,
+};
+
+#[derive(Debug, From, Serialize)]
+pub enum Event<P> {
+    // ========== Events triggered by the libp2p network behavior ==========
```

Comment:
Can't we just wrap all of them and pass the libp2p event through wholesale? I'm not big on introducing abstraction here.

Reply:
Can you explain why not?  I think the more we can keep libp2p types out of our public interfaces the better.

Reply:
Component events aren't entirely public, if Rust supported it, they'd have 100% private variants. No one but the component should be instantiating them directly anyway.

Reply:
> Component events aren't entirely public
Technically, they are :)

I'm really loath to do even more damage to our already fundamentally-wrong public API!